src/value.rs

```diff
@@ -845,44 +845,8 @@ impl Iterator for CompactBitsIter<'_> {
     }
 }
 
-trait Padding {
-    fn read_left_padding<I: Iterator<Item = bool>>(
-        bits: &mut I,
-        ty_l: &Final,
-        ty_r: &Final,
-    ) -> Result<(), EarlyEndOfStreamError>;
-
-    fn read_right_padding<I: Iterator<Item = bool>>(
-        bits: &mut I,
-        ty_l: &Final,
-        ty_r: &Final,
-    ) -> Result<(), EarlyEndOfStreamError>;
-}
-
-enum CompactEncoding {}
-
-impl Padding for CompactEncoding {
-    fn read_left_padding<I: Iterator<Item = bool>>(
-        _: &mut I,
-        _: &Final,
-        _: &Final,
-    ) -> Result<(), EarlyEndOfStreamError> {
-        // no padding
-        Ok(())
-    }
-
-    fn read_right_padding<I: Iterator<Item = bool>>(
-        _: &mut I,
-        _: &Final,
-        _: &Final,
-    ) -> Result<(), EarlyEndOfStreamError> {
-        // no padding
-        Ok(())
-    }
-}
-
 impl Value {
-    fn from_bits<I: Iterator<Item = u8>, P: Padding>(
+    fn from_bits<I: Iterator<Item = u8>>(
         bits: &mut BitIter<I>,
         ty: &Final,
     ) -> Result<Self, EarlyEndOfStreamError> {
```

Comment:
00729173ea8d8b3044c3c034a0b546745c2ac63e: The body of `from_bits` should be moved to `from_compact_bits` and `from_bits` should be removed.